### PR TITLE
doc: Add doc on how to write Ruby contracts

### PR DESCRIPTION
**Pull request by xxuejie**

### CODE REVIEW

**Comment by ashchan** [comment]
👍

File: docs/how-to-write-contracts.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+![Transaction Model](/docs/images/tx.png)
+
+Note that to focus on explaining contract model, certain fields in a cell(such as cell data) are omitted here for simplicity reason.
+
+In CKB, each cell has 2 contracts:
+
+* A required lock script, note cell only stores the hash of the lock script, we will explain this later. This is used to verify *who* can unlock the cell, for example, we can put secp256k1 verification in lock script to verify a signature is indeed signed by the cell owner, and only unlock the cell when the signature is valid.
+* An optional type script. This is used to verify *how* one can use the cell, for example, type script can be used to ensure that no new tokens are created from the air when transferring user defined tokens.
+
+When unlocking a cell in a transaction, the corresponding input part should contain an unlock script, the hash of the unlock script should match `lock` part in the referenced cell(in other words, we are using `P2SH` scheme here). So another way of looking at this problem here, is that it's the `lock` part in the cell that really determines what unlock script is used here, and we can treat `lock` and `unlock` here as the same thing.
```
[the remark this answers is not in the document]
Comment (xxuejie) [reply docs/how-to-write-contracts.md]: This leads back to our old discussion on whether we should move unsigned args out of Script model, @doitian thoughts?

File: docs/how-to-write-contracts.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+# Transaction Model
+
+Below is an example of CKB's transaction:
+
+![Transaction Model](/docs/images/tx.png)
+
+Note that to focus on explaining contract model, certain fields in a cell(such as cell data) are omitted here for simplicity reason.
+
+In CKB, each cell has 2 contracts:
```
[the remark this answers is not in the document]
Comment (ashchan) [reply docs/how-to-write-contracts.md]: Good point. I'm very confused by contract for smart contract, contract for mruby contract, and contract script (and `type`)...

File: docs/how-to-write-contracts.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (doitian) [reply docs/how-to-write-contracts.md]: We need to make the structure easy to explain. Update the core data structure if needed.

[the remark this answers is not in the document]
Comment (ashchan) [reply docs/how-to-write-contracts.md]: True.

**Comment by ashchan** [comment]
@janx @xxuejie Anything we can do to push this forward?

**Comment by janx** [comment]
I think it's ok to merge.

**Pull request merged**